### PR TITLE
Replace Identifiable.getProperties by getProperty & setProperty

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Change access to properties fields of Identifiable class

**What is the current behavior?** *(You can also link to an open issue here)*
Identifiable.getProperties is used to set new properties

**What is the new behavior (if this is a feature change)?**
Identifiable.getProperties method is no longer accessible, replaced by getProperty & setProperty

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-impl/src/test/groovy/com/powsybl/iidm/network/impl/IdentifiableExtensionGroovyTest.groovy [hunk 1]
```diff
@@ -44,10 +44,10 @@ class IdentifiableExtensionGroovyTest {
     @Test
     void testProperty() {
         assertFalse(s.hasProperty())
-        assertNull(s.greeting)
-        s.greeting = "hello"
```
Comment (geofjamg) [inline iidm/iidm-impl/src/test/groovy/com/powsybl/iidm/network/impl/IdentifiableExtensionGroovyTest.groovy]: Your change should not impact this line of code and Groovy automatic property creation should continue to work as in the test.

File: loadflow/loadflow-validation/src/test/java/com/powsybl/loadflow/validation/ShuntCompensatorsValidationTest.java [hunk 2]
```diff
@@ -111,12 +108,13 @@ public void checkShuntsValues() {
...
     @Test
     public void checkShunts() {
-        // “q” = - bPerSection * currentSectionCount * v^2
+        // â€œqâ€� = - bPerSection * currentSectionCount * v^2
         assertTrue(ShuntCompensatorsValidation.INSTANCE.checkShunts(shunt, strictConfig, formatterConfig, NullWriter.NULL_WRITER));
         Mockito.when(shuntTerminal.getQ()).thenReturn(171.52);
         assertFalse(ShuntCompensatorsValidation.INSTANCE.checkShunts(shunt, strictConfig, formatterConfig, NullWriter.NULL_WRITER));
 
-        // if the shunt is disconnected then either “q” is not defined or “q” is 0
+        // if the shunt is disconnected then either â€œqâ€� is not defined or â€œqâ€� is
```
Comment (mathbagu) [inline loadflow/loadflow-validation/src/test/java/com/powsybl/loadflow/validation/ShuntCompensatorsValidationTest.java]: Encoding issue

File: iidm/iidm-impl/src/test/groovy/com/powsybl/iidm/network/impl/IdentifiableExtensionGroovyTest.groovy (hunk 1 above)
Comment (tadam50) [reply iidm/iidm-impl/src/test/groovy/com/powsybl/iidm/network/impl/IdentifiableExtensionGroovyTest.groovy]: Without changes, the test failed with following message:
No signature of method: java.util.LinkedHashMap.getProperty() is applicable for argument types: (java.lang.String) values: [greeting]

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Identifiable.java [hunk 3]
```diff
@@ -35,6 +37,34 @@
 
     /**
      * Get properties associated to the object.
+     *
+     * @deprecated Use {@link #getProperty()} & {@link #setProperty()} instead.
      */
+    @Deprecated
     Properties getProperties();
```
Comment (mathbagu) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Identifiable.java]: Add a default implementation that throws an exception (UnsupportedOperationException).

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractIdentifiable.java [hunk 4]
```diff
@@ -50,6 +50,31 @@ public boolean hasProperty() {
...
+        return getProperties().setProperty(key, value);
+    }
+
+    @Override
+    public Set<String> getPropertyNames() {
+        return getProperties().stringPropertyNames();
+    }
+
     @Override
     public Properties getProperties() {
```
Comment (mathbagu) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractIdentifiable.java]: Remove this method, and allocate a HashMap or a TreeMap or a LinkedHashMap.

File: iidm/iidm-impl/src/test/groovy/com/powsybl/iidm/network/impl/IdentifiableExtensionGroovyTest.groovy [hunk 5]
```diff
@@ -43,10 +43,9 @@ class IdentifiableExtensionGroovyTest {
 
     @Test
     void testProperty() {
-        assertFalse(s.hasProperty())
```
Comment (mathbagu) [inline iidm/iidm-impl/src/test/groovy/com/powsybl/iidm/network/impl/IdentifiableExtensionGroovyTest.groovy]: Revert this change

File: iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/VariantManagerImplTest.java [hunk 6]
```diff
@@ -57,11 +57,36 @@ public boolean hasProperty() {
...
 
+        @Override
+        public boolean hasProperty(String key) {
+            return false;
+        }
+
         @Override
         public Properties getProperties() {
             return null;
         }
```
Comment (mathbagu) [inline iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/VariantManagerImplTest.java]: Do not override this method: the default behaviour is given by the interface (default method)
